### PR TITLE
Add section on cert caching

### PR DESCRIPTION
**Pull request by HaoK**
Doc perf fixes for https://github.com/dotnet/aspnetcore/issues/12324

cc @blowdart @Tratcher @JamesNK

### CODE REVIEW

File: aspnetcore/security/authentication/certauth.md [hunk 1]
```diff
@@ -551,6 +553,18 @@ namespace AspNetCoreCertificateAuthApi
 
 <a name="occ"></a>
 
+## Certificate validation cachine
```
Comment (JamesNK) [inline aspnetcore/security/authentication/certauth.md:556]: There is probably conditional display instruction to only display this content when viewing 5.0 docs, e.g.

`::: moniker range=">= aspnetcore-3.0"`

`::: moniker-end`

Should use this so we don't need to mention the version it is available from. I don't know the variable is called in 5.0

File: aspnetcore/security/authentication/certauth.md [hunk 2]
```diff
@@ -39,7 +39,9 @@ public void ConfigureServices(IServiceCollection services)
 {
     services.AddAuthentication(
         CertificateAuthenticationDefaults.AuthenticationScheme)
-        .AddCertificate();
+        .AddCertificate()
+	// Adding a ICertificateValidationCache will result in certificate auth caching the results, the default implementation uses a memory cache
```
[the remark this answers is not in the document]
Comment (JamesNK) [reply aspnetcore/security/authentication/certauth.md:43]: Adding `AddCertificateCache` to this sample makes it specific to 5.0

File: aspnetcore/security/authentication/certauth.md (hunk 1 above)
Comment (scottaddie) [reply aspnetcore/security/authentication/certauth.md:556]: As James mentioned, you can toggle the visibility of content for 5.0 and later with the following:

```
::: moniker range=">= aspnetcore-5.0"

5.0+ content here

::: moniker-end
```

File: aspnetcore/security/authentication/certauth.md [hunk 3]
```diff
@@ -551,6 +553,18 @@ namespace AspNetCoreCertificateAuthApi
 
 <a name="occ"></a>
 
+## Certificate validation cachine
+
+ASP.NET Core 5 preview 7 and later adds the ability to enable caching of validation results which drastically improves performance as validation is a expensive operation.
```
Comment (scottaddie) [inline aspnetcore/security/authentication/certauth.md:558]:
```suggestion
ASP.NET Core 5.0 Preview 7 and later adds the ability to enable caching of validation results. The caching dramatically improves performance, as validation is an expensive operation.
```

File: aspnetcore/security/authentication/certauth.md (hunk 1 above)
Comment (scottaddie) [inline aspnetcore/security/authentication/certauth.md:556]:
```suggestion
## Certificate validation caching
```

File: aspnetcore/security/authentication/certauth.md [hunk 4]
```diff
@@ -551,6 +553,18 @@ namespace AspNetCoreCertificateAuthApi
...
+
+To enable the the default implementation that uses Memory cache, just call:
+
+`.AddCertificate().AddCertificateCache(o => { o.CacheSize = 1024; o.CacheEntryExpiration = TimeSpan.FromMinutes(2); })`
+
+You also can provide in your own implementation via `services.AddSingleton<ICertificateValidationCache, YourCache>()`
+
+
+
```
Comment (scottaddie) [inline aspnetcore/security/authentication/certauth.md:567]: Remove the blank lines.

```suggestion
```

**Comment by HaoK** [comment]
Updated, how's it look now @JamesNK

File: aspnetcore/security/authentication/certauth.md [hunk 5]
```diff
@@ -551,6 +555,18 @@ namespace AspNetCoreCertificateAuthApi
 
 <a name="occ"></a>
 
+::: moniker range=">= aspnetcore-5.0"
+## Certificate validation caching
+
+ASP.NET Core 5.0 Preview 7 and later adds the ability to enable caching of validation results. The caching dramatically improves performance, as validation is an expensive operation.
```
Comment (JamesNK) [inline aspnetcore/security/authentication/certauth.md:564]:
```suggestion
ASP.NET Core 5.0 Preview 7 and later adds the ability to enable caching of validation results. The caching dramatically improves performance of certificate authentication, as validation is an expensive operation.
```

File: aspnetcore/security/authentication/certauth.md [hunk 6]
```diff
@@ -39,8 +39,11 @@ public void ConfigureServices(IServiceCollection services)
...
     services.AddAuthentication(
         CertificateAuthenticationDefaults.AuthenticationScheme)
-        .AddCertificate()
+::: moniker range="< aspnetcore-5.0"
+        .AddCertificate();
+::: moniker-end
 ::: moniker range=">= aspnetcore-5.0"
+        .AddCertificate()
         // Adding a ICertificateValidationCache will result in certificate auth caching the results, the default implementation uses a memory cache
         .AddCertificateCache()
```
Comment (JamesNK) [inline aspnetcore/security/authentication/certauth.md:48]:
```suggestion
        .AddCertificateCache();
```

**Comment by HaoK** [comment]
Look good to you @scottaddie ?

**Pull request merged**